Managers/AudioMan.h

```diff
@@ -1,707 +1,446 @@
+		/// </summary>
+		/// <param name="pitch">The desired pitch multiplier. Keep it > 0.</param>
+		/// <param name="excludeMusic">Whether to exclude the music from pitch modification</param>
+		void SetGlobalPitch(double pitch = 1.0, bool excludeMusic = false);
+
+		/// <summary>
+		/// Sets/updates the distance attenuation for a specific sound.
+		/// Will only have an effect if the sound is currently being played.
+		/// </summary>
+		/// <param name="pSound">A pointer to a Sound object. Ownership IS NOT transferred!</param>
+		/// <param name="distance">Distance attenuation scalar: 0 = full volume, 1.0 = max distant, but not completely inaudible.</param>
+		/// <returns>Whether a sample of the Sound is currently being played by any of the channels, and the attenuation was successfully set.</returns>
+		bool SetSoundAttenuation(Sound *pSound, float distance = 0.0);
+
+		/// <summary>
+		/// Sets/updates the frequency/pitch for a specific sound. 
+		/// Will only have an effect if the sound is currently being played.
+		/// </summary>
+		/// <param name="pSound">A pointer to a Sound object. Ownership IS NOT transferred!</param>
+		/// <param name="pitch">New pitch, a multiplier of the original normal frequency. Keep it > 0.</param>
+		/// <returns>Whether a sample of the Sound is currently being played by any of the channels, and the pitch was successfully set.</returns>
+		bool SetSoundPitch(Sound *pSound, float pitch);
+
+		/// <summary>
+		/// Sets/updates the frequency/pitch for the music channel.
+		/// </summary>
+		/// <param name="pitch">New pitch, a multiplier of the original normal frequency. Keep it > 0.</param>
+		/// <returns>Whether a sample of the Sound is currently being played by any of the channels, and the pitch was successfully set.</returns>
+		bool SetMusicPitch(float pitch);
+
+		/// <summary>
+		/// Gets the path of the last played music stream.
+		/// </summary>
+		/// <returns>The file path of the last played music stream.</returns>
+		std::string GetMusicPath() const { return m_MusicPath; }
+
+		/// <summary>
+		/// Gets the position of playback of the current music stream, in seconds.
+		/// </summary>
+		/// <returns>The current position of the current stream playing, in seconds.</returns>
+		double GetMusicPosition();
+
+		/// <summary>
+		/// Sets the music to a specific position in the song.
+		/// </summary>
+		/// <param name="position">The desired position from the start, in seconds.</param>
+		void SetMusicPosition(double position);
+
+		/// <summary>
+		/// Returns the number of audio channels currently used.
+		/// </summary>
+		/// <returns>The number of audio channels currently used.</returns>
+		int GetPlayingChannelCount();
+
+		/// <summary>
+		/// Returns the number of audio channels available in total.
+		/// </summary>
+		/// <returns>The number of audio channels available in total.</returns>
+		int GetTotalChannelCount();
+#pragma endregion
+
+#pragma region Playback Handling
+		/// <summary>
```

Comment:
We can't actually.
A sound file that is ogg or mp3 will crash the game on load without an error message.
Same goes for music that isn't ogg I believe.
Obviously both fmod and gorilla support the formats but our implementations don't.

Reply:
Oh really? I didn't realize that. When we do the ticket to update to new fmod we should make a note of seeing how much trouble it would be to allow other sound types.

Reply:
Honestly, the only thing we should do here is make ogg-vorbis a standard for _all_ sounds.
No need for the multi-format mess.
It's lossy, yes, but that doesn't matter too much. Also it's much more size efficient than mp3.
And if we get vorbis to work i'm pretty sure flac will work too for those times you _really need your pewpew sound to be 36864kbps_.

Reply:
We can do that as a separate issue after/alongside updating to new fmod. No sense in doing upgrades to a system that's gonna be outmoded soon.